### ISSUE TITLE
Using unicode literals as indentifier produces a weird warning about snake case

### ISSUE THREAD

**Issue by eiennohito** [rust-lang/rust#21735]
Sample code (uses Hiragana を as an identifier)
```rust
fn を(i: i32) -> i32 {
  i + 5
}
```

A warning:
```
test.rs:8:1: 10:2 warning: function `を` should have a snake case name such as `を`, #[warn(non_snake_case)] on by default
test.rs:8 fn を(i: i32) -> i32 {
test.rs:9   i + 5
test.rs:10 }
```

**Comment by nagisa** [comment]
The reason is that some scripts in Unicode (such as kana) have no associated case to them (i.e. they are case-less) and the lint only [allows lowercase codepoints](https://github.com/rust-lang/rust/blob/bedd8108dc9b79402d1ea5349d766275f73398ff/src/librustc/lint/builtin.rs#L948-955).

This is E-easy material, but it is unclear whether the lint should just outright accept any non-uppercase codepoint or we should refine the error message to not have “such as `を`” suggestion if the lint is triggered by no-case codepoints.
